Libplanet.Action/State/IBlockChainStates.cs

```diff
@@ -56,51 +56,11 @@ public interface IBlockChainStates
         /// <seealso cref="IWorldState"/>
         IWorldState GetWorldState(HashDigest<SHA256>? stateRootHash);
 
-        /// <summary>
-        /// Returns the <see cref="IAccountState"/> in the <see cref="BlockChain"/>
-        /// at <paramref name="offset"/>.
-        /// </summary>
-        /// <param name="offset">The <see cref="BlockHash"/> of the <see cref="Block"/> to create
-        /// for which to create an <see cref="IAccountState"/>.</param>
-        /// <param name="address">The <see cref="Address"/> of <see cref="IAccountState"/>
-        /// to be returned.</param>
-        /// <returns>
-        /// The <see cref="IAccountState"/> at <paramref name="offset"/>.
-        /// </returns>
-        /// <exception cref="ArgumentException">Thrown when <paramref name="offset"/> is not
-        /// <see langword="null"/> and one of the following is true.
-        /// <list type="bullet">
-        ///     <item><description>
-        ///         Corresponding <see cref="Block"/> is not found in the <see cref="IStore"/>.
-        ///     </description></item>
-        ///     <item><description>
-        ///         Corresponding <see cref="Block"/> is found but its state root is not found
-        ///         in the <see cref="IStateStore"/>.
-        ///     </description></item>
-        /// </list>
-        /// </exception>
-        /// <seealso cref="IAccountState"/>
-        IAccountState GetAccountState(BlockHash? offset, Address address);
-
-        /// <summary>
-        /// Returns the <see cref="IAccountState"/> in the <see cref="BlockChain"/>
-        /// of root hash <paramref name="stateRootHash"/>.
-        /// </summary>
-        /// <param name="stateRootHash">The <see cref="HashDigest{SHA256}"/> of the root hash
-        /// for which to create an <see cref="IAccountState"/>.</param>
-        /// <returns>
-        /// The <see cref="IAccountState"/> of state root hash <paramref name="stateRootHash"/>.
-        /// </returns>
-        /// <seealso cref="IAccountState"/>
-        IAccountState GetAccountState(HashDigest<SHA256>? stateRootHash);
-
         /// <summary>
         /// Gets a state associated to specified <paramref name="address"/>.
         /// </summary>
         /// <param name="offset">The <see cref="BlockHash"/> of the <see cref="Block"/> to fetch
         /// the state from.</param>
-        /// <param name="accountAddress">The <see cref="Address"/> of the account to fetch
```

Comment:
Maybe we should mentions about legacyState I guess...

Reply:
Specs haven't been finalized. We might get rid of access to "lower level" values altogether. I'm making minor tweaks for now to see if it is viable. 😗